app/controllers/stripe-setup/director/post.controller.js

```diff
@@ -0,0 +1,168 @@
+const logger = require('../../../utils/logger')(__filename)
+const paths = require('../../../paths')
+const formatAccountPathsFor = require('../../../utils/format-account-paths-for')
+const { isSwitchingCredentialsRoute, isAdditionalKycDataRoute, getCurrentCredential, getSwitchingCredential } = require('../../../utils/credentials')
+const { response } = require('../../../utils/response')
+const { validateMandatoryField, validateEmail } = require('../../../utils/validation/server-side-form-validations')
+const { listPersons, updateDirector, createDirector, updateCompany } = require('../../../services/clients/stripe/stripe.client')
+const { validateDateOfBirth } = require('../../../utils/validation/server-side-form-validations')
+const { ConnectorClient } = require('../../../services/clients/connector.client')
+const connector = new ConnectorClient(process.env.CONNECTOR_URL)
+const FIRST_NAME_FIELD = 'first-name'
+const LAST_NAME_FIELD = 'last-name'
+const EMAIL_FIELD = 'email'
+const DOB_DAY_FIELD = 'dob-day'
+const DOB_MONTH_FIELD = 'dob-month'
+const DOB_YEAR_FIELD = 'dob-year'
+const listOfFields = [FIRST_NAME_FIELD, LAST_NAME_FIELD, EMAIL_FIELD, DOB_DAY_FIELD, DOB_MONTH_FIELD, DOB_YEAR_FIELD]
+const validationRules = [
+  {
+    field: FIRST_NAME_FIELD,
+    validator: validateMandatoryField,
+    maxLength: 100
+  },
+  {
+    field: LAST_NAME_FIELD,
+    validator: validateMandatoryField,
+    maxLength: 100
+  },
+  {
+    field: EMAIL_FIELD,
+    validator: validateEmail,
+    maxLength: 320
+  }
+]
+const trimField = (key, store) => lodash.get(store, key, '').trim()
+
+module.exports = async function (req, res, next) {
+  const isSwitchingCredentials = isSwitchingCredentialsRoute(req)
+  const collectingAdditionalKycData = isAdditionalKycDataRoute(req)
+  const currentCredential = getCurrentCredential(req.account)
+
+  const stripeAccountSetup = req.account.connectorGatewayAccountStripeProgress
+
+  if (!stripeAccountSetup) {
+    return next(new Error('Stripe setup progress is not available on request'))
+  }
+  if (stripeAccountSetup.director) {
+    req.flash('genericError', 'You’ve already provided director details. Contact GOV.UK Pay support if you need to change them.')
+    return res.redirect(303, formatAccountPathsFor(paths.account.dashboard.index, req.account.external_id))
+  }
+
+  const formFields = getFormFields(req.body, listOfFields)
+  const errors = validateDirector(req.body, formFields)
+
+  if (!lodash.isEmpty(errors)) {
+    const pageData = {
+      firstName: formFields[FIRST_NAME_FIELD],
+      lastName: formFields[LAST_NAME_FIELD],
+      email: formFields[EMAIL_FIELD],
+      dobDay: formFields[DOB_DAY_FIELD],
+      dobMonth: formFields[DOB_MONTH_FIELD],
+      dobYear: formFields[DOB_YEAR_FIELD]
+    }
+
+    pageData['errors'] = errors
+
+    return response(req, res, 'stripe-setup/director/index', {
+      ...pageData, isSwitchingCredentials, collectingAdditionalKycData, currentCredential
+    })
+  } else {
```

Comment:
As we return if there are errors we could remove the `else` to reduce nesting. Not something that needs to be fixed, just a personal preference for readability